tests/FunctionCallTest.php

```diff
@@ -3181,6 +3181,18 @@ function foo(never $_): void
                 'ignored_issues' => [],
                 'php_version' => '8.1',
             ],
+            'disallowVoidNeverTypeForParam' => [
+                'code' => '<?php
+                    /**
+                     * this is just one example where psalm uses isVoid and isNever, which is why void|never should not be collapsed to null
+                     * @param void|never $i
```

Comment:
This should result in `void` (as `never` is the bottom type) and then trigger `ReservedWord`